clang/lib/Format/UnwrappedLineParser.cpp

```diff
@@ -2371,7 +2371,8 @@ bool UnwrappedLineParser::tryToParseLambdaIntroducer() {
   if ((Previous && ((Previous->Tok.getIdentifierInfo() &&
                      !Previous->isOneOf(tok::kw_return, tok::kw_co_await,
                                         tok::kw_co_yield, tok::kw_co_return)) ||
-                    Previous->closesScope())) ||
+                    (Previous->closesScope() &&
+                     !Previous->endsSequence(tok::r_paren, tok::greater)))) ||
```

Comment:
And if it is casted to `int(*)`?

I know we didn't annotate stuff here, it is hard to detect if this is a cast, but is there maybe a different way which would be a bit more resilient?

Reply:
I've added the handling of casting to a function pointer, the only other case I can think of.